db/models.py

```diff
@@ -50,3 +62,70 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="orders",
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        date = self.created_at.strftime("%Y-%m-%d %H:%M:%S")
+        return date
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        MovieSession,
+        on_delete=models.CASCADE,
+        related_name="sessions_tickets"
+    )
+    order = models.ForeignKey(
+        Order,
+        on_delete=models.CASCADE,
+        related_name="orders_tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_ticket_seat_session_row"
+            )
+        ]
+
+    def __str__(self) -> str:
+        return (f"{self.movie_session.movie.title} "
+                f"{self.movie_session.show_time} "
+                f"(row: {self.row}, seat: {self.seat})")
+
+    def clean(self) -> None:
+        if not (1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row):
+            raise ValidationError(
+                {"seat": f"seat number must be in "
+                 f"available range: (1, seats_in_row): (1, "
+                 f"{self.movie_session.cinema_hall.seats_in_row})"}
+            )
+
+        if not (1 <= self.row <= self.movie_session.cinema_hall.rows):
+            raise ValidationError(
+                {"row": f"row number must be in "
+                    f"available range: (1, rows): (1, "
+                    f"{self.movie_session.cinema_hall.rows})"}
+            )
+
+    def save(self, *args, **kwargs) -> Any:
+        self.full_clean()
+        return super().save(*args, **kwargs)
```

Comment:
Are you sure it returns it? It returns None actually(you can hover your mouse on super.save and will see which type it should return